### DISCUSSION TITLE
Create an autoconfig to facilitate development

### DISCUSSION THREAD

**Discussion by marcosstefani** [sqlalchemy/sqlalchemy#11131] [Ideas]
I propose an alternative in the way of using the SQLAlchemy library, offering several advantages:

Advantages:
- Code simplification: Reducing the amount of boilerplate code makes the code more concise and readable.
- Increased productivity: Allowing developers to focus on business logic without worrying about database access details.
- Enhanced maintainability: Facilitating code maintenance by centralizing data access logic in one place.
- Code reuse: Allowing code reuse across different projects, increasing development efficiency.

I offer in this repository https://github.com/marcosstefani/sqlalchemy-autoconfig in the file `app_example.py` a demonstration of the usage I propose for having more efficient code using SQLAlchemy. The code structure is organized, and library methods are being used more correctly in terms of programming best practices.

I haven't created the PR for your project because I got lost in terms of dependencies. What I've done depends on the PyYAML library, and I genuinely didn't see how to handle that in the project's source code.

I am available to assist you with any questions or needs related to this.

**Comment by CaselIT** [comment]
Hi,

I think something like that can be an external library, similar to flask-sqlalchemy.

Just a note after skimming the code, session is not a global object and using it like it is will not work on any realistic program.

**Reply by marcosstefani** [reply]
Could be. But Flask-SQLAlchemy is specific for Flask applications and this is not.

**Reply by CaselIT** [reply]
Sure, that was just an example of 3-rd party library that does something somewhat comparable
